c2cgeoportal/views/layers.py

```diff
@@ -244,8 +245,16 @@ def security_cb(r, feature, o):
                 if allowed.scalar() == 0:
                     raise HTTPForbidden()
 
-        protocol = self._get_protocol_for_layer(layer, before_create=security_cb)
-        return protocol.create(self.request)
+                # check if geometry is valid
+                self.validate_geometry_(spatial_elt)
+
+        protocol = self._get_protocol_for_layer(layer, before_create=check_geometry)
+        try:
+            features = protocol.create(self.request)
+            return features
```

Comment:
It's not necessary, but I wanted to make it clear that `protocol.create()` returns a list of features.